src/io/mwaf.rs

```diff
@@ -171,50 +178,85 @@ impl FlagFileSet {
         let first_timestep = timestep_range.start;
         let gps_start = corr_ctx.timesteps[first_timestep].gps_time_ms as f64 / 1e3
             + corr_ctx.metafits_context.corr_int_time_ms as f64 / 2e3;
-        let num_rows = timestep_range.len() * corr_ctx.metafits_context.num_baselines;
+
+        let ant_pairs = vis_sel.get_ant_pairs(&corr_ctx.metafits_context);
+        let mut autos_included = false;
+        for &(a, b) in &ant_pairs {
+            if a == b {
+                autos_included = true;
+                break;
+            }
+        }
+
+        let unique_ant_indices = ant_pairs
```

Comment:
I like that you're guarding against non-unique ant pairs, and I should probably be doing the same elsewhere in the code, which means it might make sense to add `vis_ctx.get_unique_aint_pairs()`. only issue is, this implementation re-orders the ant pairs, which could also cause problems. 🤔

Reply:
ohh, you're ordering ant indices, not ant pairs, never mind.